src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AutoInsert/AutoClosingTagOnAutoInsertProvider.cs

```diff
@@ -234,9 +234,25 @@ private static bool TryEnsureOwner_WorkaroundCompilerQuirks(int afterCloseAngleI
                 //
                 // The owner will be the </div>. Note this does not happen outside of C# blocks.
 
-                var closeAngleIndex = afterCloseAngleIndex - 1;
-                var closeAngleSourceChange = new SourceChange(closeAngleIndex, length: 0, newText: string.Empty);
+                var closeAngleSourceChange = new SourceChange(afterCloseAngleIndex - 1, length: 0, newText: string.Empty);
                 currentOwner = syntaxTree.Root.LocateOwner(closeAngleSourceChange);
+
+                // Get the real closing angle if we get the quote from an attribute syntax. See https://github.com/dotnet/razor-tooling/issues/5694
+                switch (currentOwner)
+                {
+                    case MarkupTextLiteralSyntax { Parent: MarkupAttributeBlockSyntax { Parent: MarkupStartTagSyntax startTag } }:
+                        currentOwner = startTag.CloseAngle;
+                        break;
+                    case MarkupTextLiteralSyntax { Parent: MarkupMiscAttributeContentSyntax { Parent: MarkupStartTagSyntax startTag } }:
+                        currentOwner = startTag.CloseAngle;
+                        break;
+                    case MarkupTextLiteralSyntax { Parent: MarkupTagHelperAttributeSyntax { Parent: MarkupTagHelperStartTagSyntax startTagHelper } }:
+                        currentOwner = startTagHelper.CloseAngle;
+                        break;
+                    case MarkupTextLiteralSyntax { Parent: MarkupMiscAttributeContentSyntax { Parent: MarkupTagHelperStartTagSyntax startTagHelper } }:
+                        currentOwner = startTagHelper.CloseAngle;
+                        break;
+                }
```

Comment:
Could we also add [minimized attributes](https://github.com/dotnet/razor-compiler/blob/ba467bb6c696f1dcd259763fe1599dffdd483f89/src/Microsoft.AspNetCore.Razor.Language/src/Syntax/Syntax.xml#L68) for HTML/TagHelpers to this list?